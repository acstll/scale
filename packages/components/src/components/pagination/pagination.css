.pagination {
  height: 20px;
  width: 200px;
}
.pagination ul {
  display: inline-block;
  position: relative;
  margin: 0;
  padding: 0;
  list-style: none;
  cursor: default;
}

.pagination li {
  position: relative;
  display: block;
  float: left;
  padding: 0;
  margin-right: 0.5rem;
  margin-left: 0;
  width: 0.8rem;
  height: 0.8rem;
  cursor: pointer;
  border-radius: 50%;
  border: solid 1px black;
}

.pagination li a {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  outline: none;
  border-radius: 50%;
  background-color: white;
  background-color: rgba(255, 255, 255, 0.3);
  text-indent: -999em;
  cursor: pointer; /* make the text accessible to screen readers */
  position: absolute;
}

.fillup li a {
  overflow: hidden;
  background-color: rgba(0, 0, 0, 0);
  box-shadow: inset 0 0 0 2px rgba(255, 255, 255, 1);
  transition: background 0.3s;
}

.fillup li a::after {
  content: '';
  position: absolute;
  bottom: 0;
  height: 0;
  left: 0;
  width: 100%;
  background-color: #fff;
  box-shadow: 0 0 1px #fff;
  transition: height 0.3s;
}

.fillup li a:hover,
.fillup li a:focus {
  background-color: #e20074;
}

.fillup li.current a::after {
  height: 100%;
}
