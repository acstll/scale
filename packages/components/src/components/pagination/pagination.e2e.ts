import { newE2EPage } from '@stencil/core/testing';

describe('t-pagination', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<t-pagination>this is a tag</t-pagination>');
    const element = await page.find('t-pagination');
    expect(element).toHaveClass('hydrated');
  });
});
