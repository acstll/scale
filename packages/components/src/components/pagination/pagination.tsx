import { Component, h, Prop } from '@stencil/core';
import { CssClassMap } from '../../utils/utils';
import classNames from 'classnames';

@Component({
  tag: 't-pagination',
  styleUrl: 'pagination.css',
  shadow: true,
})
export class Pagination {
  /** (optional) Tag class */
  @Prop() public customClass?: string = '';
  /** (optional) Tag theme */
  @Prop() public theme?: string = '';

  public render() {
    return (
      <div class={this.getCssClassMap()}>
        <ul class="fillup">
          <li class="active">
            <a href="#">Home</a>
          </li>
          <li>
            <a href="#">About</a>
          </li>
          <li>
            <a href="#">Products</a>
          </li>
          <li>
            <a href="#">Contact</a>
          </li>
        </ul>
      </div>
    );
  }

  private getCssClassMap(): CssClassMap {
    return classNames(
      'pagination',
      this.customClass && this.customClass,
      this.theme && `pagination--theme-${this.theme}`
    );
  }
}
