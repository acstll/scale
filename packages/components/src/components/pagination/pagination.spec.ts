import { newSpecPage } from '@stencil/core/testing';
import { Pagination } from './pagination';

describe('Pagination', () => {
  let element;
  beforeEach(async () => {
    element = new Pagination();
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [Pagination],
      html: `<t-pagination>Label</t-pagination>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should handle a custom css class', () => {
    element.customClass = 'custom';
    expect(element.getCssClassMap()).toContain('custom');
  });

  it('should handle theme css class', () => {
    element.theme = 'default';
    expect(element.getCssClassMap()).toContain('pagination--theme-default');
  });

  it('should have a default css class', () => {
    expect(element.getCssClassMap()).toContain('pagination');
  });
});
