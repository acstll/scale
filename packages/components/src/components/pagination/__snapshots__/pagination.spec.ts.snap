// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should match snapshot 1`] = `
<t-pagination>
  <mock:shadow-root>
    <div class="pagination">
      <ul class="fillup">
        <li class="active">
          <a href="#">
            Home
          </a>
        </li>
        <li>
          <a href="#">
            About
          </a>
        </li>
        <li>
          <a href="#">
            Products
          </a>
        </li>
        <li>
          <a href="#">
            Contact
          </a>
        </li>
      </ul>
    </div>
  </mock:shadow-root>
  Label
</t-pagination>
`;
